lib/drivers/stdio.ts

```diff
@@ -14,7 +14,15 @@ class StdioDriver implements IDriver {
     private _outstandingRequests = new Map<number, AsyncSubject<any>>();
     private _projectPath: string;
     private _serverPath: string;
-    public currentState: DriverState = DriverState.Disconnected;
+    private _currentState: DriverState = DriverState.Disconnected;
+    public get currentState() { return this._currentState; }
+    public set currentState(value) {
+        // Prevent the client from leaving the error state
+        // Error means bad things, like mono is working right
+        if (this._currentState !== DriverState.Error) {
```

Comment:
This is up for debate, I feel like the client should not be able to leave the error state once it enters.  Usually it means something really bad happened that it can't recover from.